From 886adf664b958aa28ccc6b06c552764069293f83 Mon Sep 17 00:00:00 2001
From: Raymond Yau <superquad.vortex2@gmail.com>
Date: Sat, 26 Feb 2011 09:24:43 +0800
Subject: [PATCH] Fix alsa-lib to regard 3D control - Depth as Playback
 Controls

This allow those mixer or recording application to find those controls which
are really used for capturing.

Signed-off-by: Raymond Yau <superquad.vortex2@gmail.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 src/mixer/simple_none.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/mixer/simple_none.c b/src/mixer/simple_none.c
index 426f2d7..395e4d6 100644
--- a/src/mixer/simple_none.c
+++ b/src/mixer/simple_none.c
@@ -937,7 +937,12 @@ static int base_len(const char *name, selem_ctl_type_t *type)
 		*type = CTL_CAPTURE_ROUTE;
 		return strlen(name);
 	}
-
+	if (strstr(name, "3D Control") > 0) {
+		if (strstr(name, "Depth") > 0) {
+			*type = CTL_PLAYBACK_VOLUME;
+			return strlen(name);
+		}
+	}
 	return 0;
 }
 
-- 
1.7.6

